### PR TITLE
Split '?' when extracting ID with HTTP URLs

### LINKED ISSUE #323: `Spotify.artist_albums` acts as `Spotify.artist` when passing HTTP artist URL
When `Spotify.artist_albums` is given an HTTP artist URL like `https://open.spotify.com/artist/7oPftvlwr6VrsViSDV7fJY?si=M3PrzRC4TBOZu8YyLYc-tA`, it behaves same as `Spotify.artist` and instead of returning the albums, it just returns artist info. However, it works fine if artist URI is passed like `spotify:artist:7oPftvlwr6VrsViSDV7fJY` and then it returns albums as expected.

For example:

```python
>>> import spotipy
>>> import spotipy.oauth2 as oauth2

>>> credentials = oauth2.SpotifyClientCredentials(
...         client_id=CLIENT_ID,
...         client_secret=CLIENT_SECRET)

>>> token = credentials.get_access_token()
>>> spotify = spotipy.Spotify(auth=token)

>>> greenday_http = 'https://open.spotify.com/artist/7oPftvlwr6VrsViSDV7fJY?si=M3PrzRC4TBOZu8YyLYc-tA'
>>> artist_as_http = spotify.artist(greenday_http)

>>> greenday_uri = 'spotify:artist:7oPftvlwr6VrsViSDV7fJY'
>>> artist_as_uri = spotify.artist(greenday_uri)

>>> artist_as_http == artist_as_uri
True

>>> artist_albums_as_http = spotify.artist_albums(greenday_http)
# `Spotify.artist_albums` acts same as `Spotify.artist`
>>> artist_albums_as_http == artist_as_http
True

>>> artist_albums_as_uri = spotify.artist_albums(greenday_uri)
# `Spotify.artist_albums` does not act same as `Spotify.artist` as expected
>>> artist_albums_as_uri == artist_as_http
False
```

### PR DESCRIPTION
**Pull request by ritiek**
Closes #323.

### CODE REVIEW

File: spotipy/client.py [hunk 1]
```diff
@@ -1049,7 +1049,8 @@ def _get_id(self, type, id):
             if type != itype:
                 self._warn('expected id of type %s but found type %s %s',
                            type, itype, id)
-            return fields[-1]
+            field, *_ = fields[-1].split('?')
```
Comment (42B) [inline spotipy/client.py]: This isn't valid syntax for Python 2.

```python
$ python
Python 3.7.1 (default, Oct 25 2018, 21:25:21) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a, *b = [1, 2, 3]
>>> a
1
>>> b
[2, 3]

$ python2
Python 2.7.15rc1 (default, Nov 12 2018, 14:31:15) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a, *b = [1, 2, 3]
  File "<stdin>", line 1
    a, *b = [1, 2, 3]
       ^
SyntaxError: invalid syntax

```

Comment (ritiek) [reply spotipy/client.py]: That's right. I'll fix it. Thanks!

**Comment by Harrison97** [comment]
Hey! I am hosting a new master branch for this repo since the old creator has been MIA for 2 years. Hoping to get this repo back on it's feet and in good standing. I've been working on updating tests and adding all new functionality and have merged some of the PR's that I thought were sufficient for merging. If your PR didn't make it, resubmit it please.

Thanks!

https://github.com/Harrison97/spotipy

**Comment by ritiek** [comment]
Superseded by [Harrison97/spotipy#3](https://github.com/Harrison97/spotipy/pull/3).

**Pull request closed without merging**